searvey/usgs.py

```diff
@@ -126,8 +126,8 @@ def normalize_usgs_stations(df: pd.DataFrame) -> gpd.GeoDataFrame:
 
     param_dict = _get_usgs_output_info().set_index("parameter_cd").to_dict()
 
-    df.end_date = pd.to_datetime(df.end_date, errors="coerce")
-    df.begin_date = pd.to_datetime(df.begin_date, errors="coerce")
+    df.end_date = pd.to_datetime(df.end_date, errors="coerce", format="mixed")
+    df.begin_date = pd.to_datetime(df.begin_date, errors="coerce", format="mixed")
```

Comment:
I think this is not the proper fix. 

The `format` argument does exist in pandas 1.X but `mixed` is not a special value in 1.X. `format="mixed"` was only added in 2.0. Consequently, even though `pd.to_datetime()` does not raise an Exception in pandas 1.X it always returns `NaT`.

For example this will raise an Assertion in 2.X but not in 1.X
``` python
import pandas as pd
assert pd.to_datetime("2001-09-23", errors="coerce", format="mixed") is pd.NaT
```

I did spend too much time on this, but I can't find a cleaner solution than checking the pandas version with `importlib.metadata.version()`

https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.to_datetime.html
https://pandas.pydata.org/pandas-docs/version/2.0/reference/api/pandas.to_datetime.html

Reply:
I pushed a fix. Not sure if it is the best solution though. If anyone things of something better, please let me know

Reply:
@pmav99 thanks for noticing this. I think what you did makes the most sense.

Reply:
I am not sure what exactly we need here, but if it to test whether the time stamp is not available or `pd.Nat` this should work for both versions:

```python
if pd.isna(pd.to_datetime("NaN")):
     do something
```

Reply:
It's actually the opposite. We don't want to detect `NaT` but to avoid generating `NaT`s in the first place. In order to do so we need an extra argument in `pd.to_datetime()` but the value of this argument is not valid in 1.X. That's why we need to detect the pandas version and conditionally pass this argument.